### ISSUE TITLE
hyperparams for training on 1200 labels

### ISSUE THREAD

**Issue by fengtingl** [dvl-tum/ciagan#5]
Hi, I enjoyed the paper and was able to run the training example. 
I set up the CelebA dataset with 1200 identities provided in legit_indices.npy. However, I am not sure about the hyper-parameter setting:

1. The learning rate is 0.0001 in train.py but 0.00001 in the paper(section 4.1)
2. The iterations of the critic, generator, and siamese are 5, 1, 1 in train.py, and 1, 3, 1 in run_training.py.
3. filter number, batch size, etc

 Could you give me the detailed hyper-parameters for training on the dataset with 1200 identities? Thanks!

The current hyper-parameters I'm using, which yield an unsatisfactory result:
'TRAIN_PARAMS': {
            'ARCH_NUM': 'unet_flex',
            'ARCH_SIAM': 'resnet_siam',
            'EPOCH_START': 0,
            'EPOCHS_NUM': 120,
            'LEARNING_RATE': 0.00001,
            'FILTER_NUM': 32,
            'ITER_CRITIC': 1,
            'ITER_GENERATOR': 3,
            'ITER_SIAMESE': 1,
	    'GAN_TYPE': 'lsgan',
            'FLAG_SIAM_MASK': False,
    },
    'DATA_PARAMS':{
        'LABEL_NUM': 1200,
        'WORKERS_NUM': 4,
        'BATCH_SIZE': 32,
        'IMG_SIZE': 128,
        'FLAG_DATA_AUGM': True,
    },
    'OUTPUT_PARAMS': {
        'SAVE_EPOCH': 1,
        'SAVE_CHECKPOINT': 60,
        'LOG_ITER': 2,
        'COMMENT': "Something here",
        'EXP_TRY': 'check',
    }

**Comment by DongzeLi-CASIA** [comment]
I have the similar problem , have you find the proper params yet?

**Comment by fengtingl** [comment]
Not yet.
For the best result so far, I use a Proxy-NCA pre-trained network, which achieves a recall@1 rate of around 81%,  as the identity discriminator. Then I extend the epochs to 300. Batch size and learning rate are set to be 16 and 1e-5, respectively. The visual quality is not satisfactory, with an FID rate of around 5.5, though. Keep me posted if you have better results.

**Comment by skye95git** [comment]
Hello, I'm also trying to train with 1200 identities，but I have a few problems：
1. There are 1563 pieces of data in legit_indices.npy, how do you get the 1,200 identities the author refers to?
2. How do you generate new images using a trained model?
3. I run the training example, but the result is not satisfactory. Is it the lack of training data?
![Aunet_flex_Dceleba_Tcheck_ep02001_est](https://user-images.githubusercontent.com/41561936/102323250-32e16a80-3fbb-11eb-84fa-0b12ba2e0896.png)

**Comment by fengtingl** [comment]
1. I tried two initializations, with the first 1200 identities and a random sampled 1200 identities. The results are not very much distinguishable.
2. I simply change the forward conditional information into a random one-hot vector. 
3. You could try the following options, but I'm not sure if there could be more improvements with more training data.
   (1) since the training of a GAN is not stable, try multiple initializations. 
   (2) use larger training epochs.

> Hello, I'm also trying to train with 1200 identities，but I have a few problems：
> 
> 1. There are 1563 pieces of data in legit_indices.npy, how do you get the 1,200 identities the author refers to?
> 2. How do you generate new images using a trained model?
> 3. I run the training example, but the result is not satisfactory. Is it the lack of training data?
>    ![Aunet_flex_Dceleba_Tcheck_ep02001_est](https://user-images.githubusercontent.com/41561936/102323250-32e16a80-3fbb-11eb-84fa-0b12ba2e0896.png)

**Comment by TheRevanchist** [comment]
What batch size you are using? We used batch size of 8 and 16 for the results in the paper.

We'll try to put the configuration and an example for the generation before the holiday, but slightly busy with current work.

**Comment by fengtingl** [comment]
Thanks for your reply.
I use batch size 16 and learning rate 1e-5 here. Could you provide a pre-trained model?
________________________________

**Comment by skye95git** [comment]
> 1. I tried two initializations, with the first 1200 identities and a random sampled 1200 identities. The results are not very much distinguishable.
> 2. I simply change the forward conditional information into a random one-hot vector.
> 3. You could try the following options, but I'm not sure if there could be more improvements with more training data.
>    (1) since the training of a GAN is not stable, try multiple initializations.
>    (2) use larger training epochs.
> 
> > Hello, I'm also trying to train with 1200 identities，but I have a few problems：
> > 
> > 1. There are 1563 pieces of data in legit_indices.npy, how do you get the 1,200 identities the author refers to?
> > 2. How do you generate new images using a trained model?
> > 3. I run the training example, but the result is not satisfactory. Is it the lack of training data?
> >    ![Aunet_flex_Dceleba_Tcheck_ep02001_est](https://user-images.githubusercontent.com/41561936/102323250-32e16a80-3fbb-11eb-84fa-0b12ba2e0896.png)

Thanks for your reply. I don't understand how to change the forward conditional information into a random one-hot vector. I have tried many times to load the trained model to generate new images, but all reported errors. Could you provide an example or share your code? Thanks.
